### PR TITLE
fix: #29909

### LINKED ISSUE #29909: Problème de conservation des filtres dans la liste des propal depuis un tiers
### Bug

J’ai besoin d’accéder à mes PR ouvertes sur un tiers mais je me retrouve avec une PR qui n’a rien à voir dans ma recherche. 
Voici mon parcours : 

Image

Image

Image

Image

### Dolibarr Version

_No response_

### Environment PHP

_No response_

### Environment Database

_No response_

### Steps to reproduce the behavior and expected behavior

_No response_

### Attached files

_No response_

### PR DESCRIPTION
**Pull request by FHenry**
fix: #29909

### CODE REVIEW

File: htdocs/comm/propal/list.php [hunk 1]
```diff
@@ -1230,7 +1232,7 @@
 }
 if (!empty($arrayfields['s.nom']['checked'])) {
 	print '<td class="liste_titre" align="left">';
-	print '<input class="flat maxwidth100" type="text" name="search_societe" value="'.dol_escape_htmltag($search_societe).'">';
+	print '<input class="flat maxwidth100" type="text" name="search_societe" value="'.dol_escape_htmltag($search_societe).'"'.($socid > 0 ? " disabled" : "").'>';
```
Comment (eldy) [inline htdocs/comm/propal/list.php:1235]: If field is disabled, the new search/submit of form will lost the filter (field will not be propagated).

File: htdocs/comm/propal/card.php [hunk 2]
```diff
@@ -2849,8 +2849,8 @@
 		}
 
 		$tmparray = $object->getTotalWeightVolume();
-		$totalWeight = $tmparray['weight'];
-		$totalVolume = $tmparray['volume'];
+		$totalWeight = isset($tmparray['weight']) ?:0;
```
Comment (eldy) [inline htdocs/comm/propal/card.php:2852]: Using isset with ?: is strange. It means it will return true if isset true instead of $weight

**Review by eldy — CHANGES_REQUESTED** [review]

File: htdocs/comm/propal/list.php (hunk 1 above)
Comment (FHenry) [reply htdocs/comm/propal/list.php:1235]: Yes exactly ,as it is done in facture/list.php.

When "filter" is send in QueryString with socid attribute, it is priorized with socid (the search_societe is displayed for the UX but not used anymore)

File: htdocs/comm/propal/card.php (hunk 2 above)
Comment (FHenry) [reply htdocs/comm/propal/card.php]: Yes, error fixed

**Pull request merged**